### DISCUSSION TITLE
Applying Mapbox view padding (camera vanishing point offset) to Deck.gl

### DISCUSSION THREAD

**Discussion by cafca** [visgl/deck.gl#5359] [Q&A]
I am using a Mapbox map with a Deck.gl overlay, which I synchronize by setting `viewState` whenever the Mapbox map moves. 

Mapbox allows [offsetting the camera vanishing point](https://docs.mapbox.com/mapbox-gl-js/example/offset-vanishing-point-with-padding/), which is very useful when toggling a sidebar. I have not been successful at synchronizing the Deck.gl overlay to this camera change. My guess is that I need to apply Mapbox' projection matrix in addition to updating the view state properties but I am not able to access and apply the right matrix.

Can somebody help me out with this? I don't know how important Mapbox compatibility is for Deck.gl but this may even be a candidate for a Deck.gl feature request.

### Demo

Notice how in the video below, the arc layer gets out of sync with the marker positions when the sidebar is opened:

https://user-images.githubusercontent.com/278463/105163717-8ee86300-5b14-11eb-9785-a9bc0e4ffbdc.mov

**Comment by Pessimistress** [comment]
> My guess is that I need to apply Mapbox' projection matrix in addition to updating the view state properties but I am not able to access and apply the right matrix.

It is possible to supply your own `projectionMatrix` as part of the view state. I have not tried it but it's probably something like `map.transform.projMatrix`.

In the long term the easiest solution is to make a feature request for deck.gl to support center offset out of the box.
